tests/foreman/ui/test_architecture.py

```diff
@@ -32,14 +34,17 @@ def test_positive_create_arch_with_os(self, test_data):
         @Assert: Architecture is created
 
         """
-        entities.OperatingSystem(name=test_data['os_name']).create()
         with Session(self.browser) as session:
-            make_arch(session, name=test_data['name'],
-                      os_names=[test_data['os_name']])
-            self.assertIsNotNone(self.architecture.search(test_data['name']))
-
-    @data(*generate_strings_list())
-    def test_positive_create_arch_with_different_names(self, name):
+            for test_data in valid_arch_os_names():
+                with self.subTest(test_data):
+                    entities.OperatingSystem(
+                        name=test_data['os_name']).create()
+                    make_arch(session, name=test_data['name'],
```

Comment:
I am sorry, but I need to repeat my question again. You are removing `html` from all `make_arch(session, name=name)` using `generate_strings_list(remove_str='html')`, but in the same time you do:
`make_arch(session, name=test_data['name'],` in first test with `{u'name': gen_string('html')`.
If all tests passed successfully that means `html` is fine or it is some kind of mystic here.
Anyway if we have situation like:
*When you use non-html characters, it is happy. When you use html characters, it stops identifying the element because characters like <, /, > are escaped in the DOM object.*
We need to introduce fix and do not do things like - `generate_strings_list(remove_str='html')` as that means that we intentionally decreasing coverage